### ISSUE TITLE
Grad, momentum size error 

### ISSUE THREAD

**Issue by goodbye61** [facebookresearch/Detectron#96]
Hi, 
I added some layers on the mask-rcnn model and then 
run a 'tools/train.net' .
Also, I tried modifying training code for fine tuning. 

I modified '_add_parameter_update_ops' function in 'optimizer.py'
and 'broadcast_parameter' in 'net.py' 

But I got an error as follows. 

-------------------------------------------------------------------------------
terminate called after throwing an instance of 'caffe2::EnforceNotMet'
terminate called after throwing an instance of 'caffe2::EnforceNotMet'
what():  [enforce fail at momentum_sgd_op.h:107] 
Input(GRAD).size() == Input(MOMENTUM).size(). 20736 vs 256 Error from operator: 
-----------------------------------------------------------------------------------

For fine tuning, What is the blind point in this case? 

Plz help me ! 

Thanks in advance :)

**Comment by rbgirshick** [comment]
As an editorial note, this type of issue report is very unlikely to bring you any help. It contains no details about what changes you made (e.g. link to a commit in a branch), so people can only guess at what you did and what problems it introduced.

**Comment by djr2015** [comment]
I'm currently trying to fine-tune RetinaNet and ran into a similar issue.
From my understanding, for each blob with weights there is a corresponding "momentum" blob (they are fed into the workspace [here](https://github.com/facebookresearch/Detectron/blob/021685d42f7e8ac097e2bcf79fecb645f211378e/lib/utils/net.py#L110)) of the same shape, containing the velocity vector used in SGD that is updated during the backward pass.

I'm guessing that if you're fine-tuning then you would have had to modify the shape of the class-specific blobs and reinitialize them to ensure consistency with the number of classes the dataset you are fine-tuning on. I solved the error you came across by making sure I applied the same procedure to the momentum blobs corresponding to the class-specific blobs.

Hope that helps!
